chef/cookbooks/network/recipes/switch_config.rb

```diff
@@ -50,7 +50,7 @@ def setup_interface(switch_config, a_node, conduit, switch_name, interface_id )
 
 switch_config={}
 
-search(:node, "*:*").each do |a_node|
+node_search_with_cache("*:*").each do |a_node|
```

Comment:
The same applies here as earlier, when we used `node_search_with_cache` with a query that does not match `/^(roles|recipes):(\w*).*$/`